pkg/api/customization/globaldns/action_globaldns.go

```diff
@@ -69,20 +69,21 @@ func (w *Wrapper) ActionHandler(actionName string, action *types.Action, request
 	for _, p := range gDNS.Spec.ProjectNames {
 		existingProjects[p] = true
 	}
+	mcAppID := convert.ToString(actionInput[client.UpdateGlobalDNSTargetsInputFieldMultiClusterAppID])
 
 	switch actionName {
 	case addProjectsAction:
-		return w.addProjects(gDNS, request, inputProjects, existingProjects)
+		return w.addProjects(gDNS, request, inputProjects, existingProjects, mcAppID)
 	case removeProjectsAction:
 		return w.removeProjects(gDNS, request, existingProjects, inputProjects)
 	default:
 		return fmt.Errorf("bad action for global dns %v", actionName)
 	}
 }
 
-func (w *Wrapper) addProjects(gDNS *v3.GlobalDNS, request *types.APIContext, inputProjects []string, existingProjects map[string]bool) error {
-	if gDNS.Spec.MultiClusterAppName != "" {
-		return fmt.Errorf("cannot add projects to globaldns as targets if multiclusterappId is set")
+func (w *Wrapper) addProjects(gDNS *v3.GlobalDNS, request *types.APIContext, inputProjects []string, existingProjects map[string]bool, mcAppID string) error {
+	if mcAppID != "" {
+		return fmt.Errorf("cannot add projects to globaldns as targets if multiclusterappID is set")
```

Comment:
We should also check if gDNS.Spec.MultiClusterAppName is not nil and error out?

Reply:
added check for mcAppName when mcAppId is not passed in actionInput